### ISSUE TITLE
More logs in failing PowerShellOnTargetMachines task

### ISSUE THREAD

**Issue by matthieu-bissat** [Microsoft/vsts-tasks#4595]
Hi,

I'm using the task "PowerShellOnTargetMachines" in Team Foundation Server 2015.

Let's imagine I have a PowerShell file with this content:
`$file = Resolve-Path "$PSScriptRoot\file-that-does-not-exist.txt"`

This script generates an error of course because the file does not exist.
When I run this file in my Windows PowerShell, I receive a nice and meaningful error:
_"Resolve-Path : Cannot find path 'C:\temp\file-that-does-not-exist.txt' because it does
not exist [blablabla and the stack trace]"_

When I run this file in a Release with PowerShellOnTargetMachines, the only message I get in the "Logs" tab of TFS is:
_"Deployment on this environment failed."_

Basically the log is useless. The error message is not returned, and if I have "Write-Verbose -verbose" statements before the failing statement they are not printed either.

Is there any way to get a more useful log?

Thanks,
Matthieu

**Comment by rajatagrawal-dev** [comment]
Hi Matthieu,

What is your task version?
Can you try adding _system.debug_ variable in your release definition and set its value to _true_? Queuing a release then might give you more useful logs.

**Comment by matthieu-bissat** [comment]
Hi, thanks for your reply!

I think our version is 1.0.25 (name of the folder on the agent machine). Well it seems this is quite an old version, do you know if there have been improvements in the "logging" area?

I tried adding System.Debug but didn't receive more logs.
However I found that log files are generated on the agent machine (in _D:\agent directory\ \_diag_) with more details that what I see in my browser. In these logs, I see the same error as the standard Powershell prompt.

Well I would still prefer seeing something in TFS, but in the meantime I can read the log files.

**Comment by rajatagrawal-dev** [comment]
Hi Matthieu,

Sorry for the lack of logs. I have confirmed that there indeed has been significant development in the logging area post version 1.0.25. I can see at least a few commits in the later versions specifically addressing this issue. If agent logs are providing relevant info then I believe that's your only workaround.
